spec/controllers/carts_controller_spec.rb

```diff
@@ -0,0 +1,11 @@
+require "rails_helper"
+
+RSpec.describe CartsController, type: :controller do
+  xit "has a session cart which is a hash" do
```

Comment:
I think the was this phrased this test belongs in a _sessions controller spec_ or _application controller spec_ as it is called by the before filter.